src/snowflake/cli/plugins/nativeapp/commands.py

```diff
@@ -391,6 +391,9 @@ def app_events(
         default="",
         help="Fetch events that are older than this time ago, in Snowflake interval syntax.",
     ),
+    limit: int = typer.Option(
+        default=0, help="Maximum number of latest events to fetch."
```

Comment:
not sure about "latest events" here. It's just a standard limit on an ordered query, so if 1000 events match and you specify limit 10, then you'll get the 10 oldest events, no?

Reply:
the query was changed to sort descending, take the limit, then sort ascending

Reply:
Am I the only one who finds that behaviour counter-intuitive? I expect truncation to happen in the natural order the results are presented in. Basically `tail | head -n` behaviour.

Reply:
I think limit in ascending order of timestamp makes sense, because if you're looking for a recent event you're likely to use the `--since` arg instead of `--limit`. Though it would be nice to have a way to get the last n events, so I'm not 100% convinced by my own argument.

Reply:
my logic is that it doesn't really make sense to get the first N events, especially in the case of a bare `snow app events --limit 100` for example. In this case, I just want to grab the latest events for my app, but not so many that they're so far in the past and no longer relevant.

I can change to `--first <N>` and `--last <N>` if that would make more sense

Reply:
I do find it surprising that `snow app events --since "2 hours" --limit 100` will not show the oldest events, personally. PL?

Reply:
Updated with `--first` and `--last` flags after offline discussion